lib/Backends/CPU/LLVMIRGen.cpp

```diff
@@ -1485,11 +1483,22 @@ void LLVMIRGen::generateLLVMIRForInstr(llvm::IRBuilder<> &builder,
       auto *slicePost = emitConstI32(builder, sliceScaleParams.post);
       auto *sliceScale = emitConstI32(builder, sliceScaleParams.scale);
 
-      createCall(builder, F,
-                 {destPtr, batchPtr, slicePtr, numSlice, sliceSize, destOffset,
-                  batchOffset, sliceOffset, batchPre, batchPost, batchScale,
-                  slicePre, slicePost, sliceScale});
+      if (sliceTy->getElementType() == ElemKind::Int8QTy) {
+        auto *F = getFunction("batchedadd", dest->getElementType());
+        createCall(builder, F,
+                   {destPtr, batchPtr, slicePtr, numSlice, sliceSize,
+                    destOffset, batchOffset, sliceOffset, batchPre, batchPost,
+                    batchScale, slicePre, slicePost, sliceScale});
+      } else {
+        auto *F = getFunction("batchedadd_i32", dest->getElementType());
+        createCall(builder, F,
+                   {destPtr, batchPtr, slicePtr, numSlice, sliceSize,
+                    destOffset, batchOffset, sliceOffset, batchPre, batchPost,
+                    batchScale, slicePre, slicePost, sliceScale});
+      }
+
```

Comment:
Could you factor the `createCall` out of the if/elseif, since it's the same either way?  Also it'd be nice to assert if `sliceTy` is neither Int8Qty not Int32QTy.